src/mongocxx/private/collection.hpp

```diff
@@ -39,6 +39,19 @@ class collection::impl {
           client_impl(client) {
     }
 
+    impl(const impl& i) : 
+      collection_t{libmongoc::collection_copy(i.collection_t)},
+      database_name{i.database_name},
+      client_impl{i.client_impl}
+    {}
+
+    impl& operator=(const impl& i) { 
+        collection_t = libmongoc::collection_copy(i.collection_t);
```

Comment:
Same thing here actually, we should clean up the original `collection_t` with `libmongoc::collection_destroy`.